### DISCUSSION TITLE
Pattern variation with crash cymbal after fill?

### DISCUSSION THREAD

**Discussion by DuskDave** [hydrogen-music/hydrogen#1880] [General]
Hi,
I was looking for a user discussion and I ended up here. Sorry if this is more for bugs and gripes, I'm just looking for 'best practice'.

I have the basic drum pattern for a song which has, say, a hi-hat on every beat. After a fill I want the hi-hat on beat 1 replaced by a crash cymbal. I could either:
1. Create a pattern with just the crash on beat 1 and play that at the same time as the main pattern but a drummer would not hit both the crash and hi-hat so it sounds unnatural.
2. Create a second pattern identical to the first with a crash instead of hi-hat on beat 1 and use that after fills.
3. Do the same as (2) but add the crash to the hi-hat mute group which mutes the hi-hat on beat 1 but cuts off the crash sound on the hi-hat for beat 2.
For a song with only one main pattern option 2 is manageable, but for songs with several different grooves each one needs a hi-hat and cymbal variant. Is there a better way? 

Thanks for any suggestions.
DD

**Comment by theGreatWhiteShark** [comment]
Hey @DuskDave ,

> For a song with only one main pattern option 2 is manageable, but for songs with several different grooves each one needs a hi-hat and cymbal variant. Is there a better way?

Good question. Whenever I create songs it always looks like I'm just drawing a diagonal in song editor.

(I don't see the benefit of option (3) over (2)) but a 4. option would be split the pattern into three patterns (one containing just the crash, one containing just the hi-hat, and the third one the remainder of the pattern) and always activate two of them. I know, it gets even more bloated. But in contrast to (2) it is much less error-prone in case you have to tweak the pattern later on.

In general you could split off common pieces of your patterns and activate multiple at the same time. Arrangements in Song editor are not that clear anymore in case the patterns are many rows apart. But since version 1.2 Hydrogen shows all notes of all patterns belonging to a single column which eases things in such setups a lot.

**Comment by DuskDave** [comment]
Thanks @theGreatWhiteShark 

I've had hydrogen installed for years but only recently used it to create a full song and yes, it has a top-left to bottom-right trend to it :) But you do get straight to the point of the matter which is trying to keep the song editor section manageable and readable.

I suppose I'm coming at this from using drum boxes like the Alesis SR-18 and others where fills always run over to the down beat of the following bar: This allows a crash cymbal to be recorded as beat 1 of the fill but played as beat 1 of the subsequent pattern. v. clever.

Weirdly, I considered your option 4 in the meantime but it does, as you point out, lead to using 2 patterns at all times which does not IMO lead to a clearer song editor.

If I could be sure that the pattern's duration was always longer than the crash's ring then I could create 2 identical hat instruments. The first used only for beat 1 shares a mute group with the crash - the second, used for all other beats is grouped with the other hats as usual. Then option 3 works better. Faster tempos would mean creating longer patterns to give the cymbal time to ring... but as a bonus longer 8-beat patterns do make the song editor narrower (if not shallower).

DD
